packages/purpose-process/src/services/purposeService.ts

```diff
@@ -336,3 +395,106 @@ const replacePurposeVersion = (
     updatedAt: newVersion.updatedAt,
   };
 };
+
+const getInvolvedTenantByEServiceMode = async (
+  eservice: EService,
+  consumerId: TenantId,
+  readModelService: ReadModelService
+): Promise<Tenant> => {
+  if (eservice.mode === eserviceMode.deliver) {
+    return retrieveTenant(consumerId, readModelService);
+  } else {
+    return retrieveTenant(eservice.producerId, readModelService);
+  }
+};
+
+const updatePurposeInternal = async (
+  purposeId: PurposeId,
+  updateContent: ApiPurposeUpdateContent | ApiReversePurposeUpdateContent,
+  organizationId: TenantId,
+  mode: EServiceMode,
+  {
+    readModelService,
+    correlationId,
+    repository,
+  }: {
+    readModelService: ReadModelService;
+    correlationId: string;
+    repository: {
+      createEvent: (createEvent: CreateEvent<PurposeEvent>) => Promise<string>;
+    };
+  }
+): Promise<{ purpose: Purpose; isRiskAnalysisValid: boolean }> => {
+  const purpose = await retrievePurpose(purposeId, readModelService);
+  assertOrganizationIsAConsumer(organizationId, purpose.data.consumerId);
+  assertPurposeIsDraft(purpose.data);
+
+  const purposeWithSameTitle = await readModelService.getSpecificPurpose(
+    purpose.data.eserviceId,
+    purpose.data.consumerId,
+    updateContent.title
+  );
+
+  if (purposeWithSameTitle) {
+    throw duplicatedPurposeTitle(updateContent.title);
+  }
```

Comment:
When the update doesn't involve a name change, this check behaves incorrectly: it retrieves a purpose with that name, but it's exactly the one we are handling. So this check has to be only triggered with the following condition: `if (updatedContent.title !== purpose.data.title)`. [Reference](https://github.com/pagopa/interop-be-monorepo/pull/177/files#diff-77a5bad1f3b877e7bfc968a9b423c16649fca358d54286d743fe34fa555564ceR611-R619).

If you wish, you could also highlight this in the tests, like `purpose update (including name change)` and `purpose update (no name change)`: in this way the bug would have been raised.

Reply:
thanks, fixed!